hail/src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -177,13 +183,23 @@ trait CodecSpec extends Serializable {
   type StagedEncoderF[T] = (Code[Region], Code[T], Code[OutputBuffer]) => Code[Unit]
   type StagedDecoderF[T] = (Code[Region], Code[InputBuffer]) => Code[T]
 
-
   def buildEncoder(t: PType, requestedType: PType): (OutputStream) => Encoder
 
   def buildEncoder(t: PType): (OutputStream) => Encoder = buildEncoder(t, t)
 
   def buildDecoder(t: PType, requestedType: PType): (InputStream) => Decoder
 
+  def encode(t: PType, region: Region, offset: Long): Array[Byte] = {
+    val baos = new ByteArrayOutputStream()
+    using(buildEncoder(t)(baos))(_.writeRegionValue(region, offset))
+    baos.toByteArray()
+  }
+
+  def decode(t: PType, bytes: Array[Byte], region: Region): Long = {
+    val bais = new ByteArrayInputStream(bytes)
+    buildDecoder(t, t)(bais).readRegionValue(region)
```

Comment:
When would `t` and `requestedType` be different? Should there be an interface that only includes one type like there is with `buildEncoder`?

Reply:
encoder needs both as well, actually.

These two differ when we're decoding a Table from disk -- the requested type is often a (smaller) supertype of the full type.